src/libostree/ostree-sysroot-cleanup.c

```diff
@@ -270,6 +270,13 @@ cleanup_old_deployments (OstreeSysroot       *self,
       g_hash_table_replace (active_boot_checksums, bootcsum, bootcsum);
     }
 
+  /* And also the staged deployment, if any */
+  if (self->staged_deployment)
+    {
+      char *deployment_path = ostree_sysroot_get_deployment_dirpath (self, self->staged_deployment);
+      g_hash_table_replace (active_deployment_dirs, deployment_path, deployment_path);
```

Comment:
Question: wanted to confirm, we don't include bootcsum here because the "staged deployment" is not ready for boot yet? ( i.e the merge and boot entries weren't ready)

Reply:
Nope, that was an oversight.  One not revealed because we weren't installing a staged deployment with a new boot checksum.

I have kind of been going back and forth on whether `/boot` should be modified during staging or finalization.  Currently it's during staging.